paper-server/src/main/java/org/bukkit/craftbukkit/block/CraftCreakingHeart.java

```diff
@@ -24,4 +33,51 @@ public CraftCreakingHeart copy() {
     public CraftCreakingHeart copy(Location location) {
         return new CraftCreakingHeart(this, location);
     }
+
+    @Override
+    public @Nullable Creaking getCreaking() {
+        return this.getBlockEntity().getCreakingProtector().map(creaking -> ((Creaking) creaking.getBukkitEntity())).orElse(null);
+    }
+
+    @Override
+    public void setCreaking(@Nullable final Creaking creaking) {
+        if (creaking == null) {
+            this.getBlockEntity().clearCreakingInfo();
+        } else {
+            Preconditions.checkArgument(this.getLocation().getWorld().equals(creaking.getLocation().getWorld()), "the location of this creaking must be in the same world than this CreakingHeart");
+            this.getBlockEntity().setCreakingInfo(((CraftCreaking) creaking).getHandle());
+        }
+    }
+
+    @Override
+    public int getCreakingRemovalDistance() {
+        return this.getSnapshot().getDistanceCreakingTooFar();
+    }
+
+    @Override
+    public void setCreakingRemovalDistance(final int distance) {
+        Preconditions.checkArgument(distance >= 0, "the distance must be >= 0");
```

Comment:
You are allowing 0, yet the codec it serialize into only allow positive integer.